packages/uniswap/src/hooks/useUniswapPrice.ts

```diff
@@ -0,0 +1,60 @@
+import { UniswapPairInterface, UNISWAP_V2_FACTORY_ADDRESS, INIT_CODE_HASH } from '../constants'
+import { compareAddress, useContractCall } from '@usedapp/core'
+import { pack, keccak256 } from '@ethersproject/solidity'
+import { getCreate2Address } from '@ethersproject/address'
+import { BigNumber, parseFixed, formatFixed } from '@ethersproject/bignumber'
+
+/**
+ * @dev
+ * function`getReserves` of UniswapV2Pair returns uint112 type,
+ * so we simply divide the reserve of quate token into the reserve of base token
+ * e.g ETH/DAI reserve of Dai / the reserve of WETH
+ * If there is no direct path or unconnected, the call returns `undefined`
+ *
+ * @param {string} baseCurrency address of base token (e.g ETH address)
+ * @param {string} quateCurrency address of quate token (e.g DAI address)
+ * @param {Object} overrides optional factory address (For Sushiswap, default Uniswap V2 Factory)
+ * @returns {FixedNumber | undefined} peek price
+ */
+export function useUniswapPrice(
+  baseCurrency: string,
+  quateCurrency: string,
+  overrides?: { factory?: string; digits?: number }
```

Comment:
Could you add the option to override the init code hash here as well? All the uniswap-v2 compatible AMMs (Sushi, pancake QS..) have different init hashes as well as different factory addresses.